controllers/challenges.js

```diff
@@ -25,7 +26,33 @@ const getChallenges = async (req, res) => {
  */
 const replaceChallenge = async (req, res) => {
     //TODO delete deprecated tag after implementing the function
-    res.status(StatusCodes.NOT_IMPLEMENTED).json({message: "Not implemented"});
+    const{ challengeID } = req.body;
+    const{ user:userID } = req.body;
+    challengeToChange = await Challenge.findById(challengeID);
+    if(!challengeToChange)
+    {
+        throw new NotFoundError(`No challenge with id : ${challengeID}`);
+    }
+    const user = await User.findById(userID);
+    if (!user) 
+    {
+        throw new NotFoundError('User does not exist');
+    }
+    if(user.challenges.indexOf(challengeToChange)===-1)
+    {
+        throw new UnauthorizedError('User not authorized to replace this challenge');
+    }
+    difficulty = challengeToChange.difficulty;
+    try{
+        const challengeReplacment = await ChallengesList.find({difficulty : difficulty});
+        await user.updateOne({ $pull: { challenges: challengeToChange } });
```

Comment:
Not sure but I believe that you have specify challenge id not an instance of a challenge to be pulled